src/js/components/homepage/visualizations/categoryMap/CategoryMap.jsx

```diff
@@ -0,0 +1,195 @@
+    }
+
+    handleWindowResize() {
+        // determine if the width changed
+        const windowWidth = window.innerWidth;
+        if (this.state.windowWidth !== windowWidth) {
+            // width changed, update the visualization width
+            this.setState({
+                windowWidth,
+                visualizationWidth: this.sectionWrapper.offsetWidth
+            });
+            if (this.props.breakdown !== null) {
+                this.buildTree(this.props.breakdown, this.props.colors, '');
+            }
+        }
+    }
+
+    buildTree(cats, colors, chosen) {
+        // put the data through d3's hierarchy system to sum and sort it
+        const root = d3.hierarchy(cats)
+        .sum((d) => (d.value))
+        .sort((d) => (d.id));
+
+        // set up a treemap object and pass in the root
+        let tileStyle = d3.treemapDice;
+        let height = 140;
+        if (this.state.windowWidth < 768) {
+            tileStyle = d3.treemapSlice;
+            height = 900;
+        }
+        const treemap = d3.treemap()
+            .round(true)
+            .tile(tileStyle)
+            .size([this.state.visualizationWidth, height])(root).leaves();
+
+        // build the tiles
+        const nodes = treemap.map((n, i) => {
+            let cell = '';
+            if (n.value !== 0) {
+                cell = (<CategoryMapCell
+                    label={n.data.name}
+                    value={n.data.value}
+                    x0={n.x0}
+                    x1={n.x1}
+                    y0={n.y0}
+                    y1={n.y1}
+                    total={n.parent.value}
+                    key={i}
+                    id={n.data.id}
+                    color={colors[i]}
+                    chosen={chosen}
+                    toggleTooltip={this.toggleTooltip} />);
+            }
+            return cell;
+        });
+
+        this.setState({
+            finalNodes: nodes
+        });
+    }
+
+    toggleTooltip(cat, value, xStart, yStart, width, height) {
+        const descSet = this.props.descriptions;
+        // find index of object item on matching cat name
+        let descIndex = '0';
+        if (cat !== 'none') {
+            descIndex = _.findIndex(descSet, { name: cat });
+        }
+
+        // set it to desc value
+        let desc = '';
+        if (cat !== 'none') {
+            desc = descSet[descIndex].value;
+        }
+
+        // set the state
+        this.setState({
+            category: cat,
+            description: desc,
+            individualValue: value,
+            x: xStart,
+            y: yStart,
+            width,
+            height
+        });
+
+        this.buildTree(this.props.breakdown, this.props.colors, this.state.category);
+    }
+
+    render() {
+        let tooltip = '';
+        let line = null;
+        if (this.state.windowWidth < 768) {
+            line = <BudgetLine />;
+        }
+        if (this.state.category !== 'none') {
```

Comment:
Can we move this into it's own function? Something like `const tooltip = this.generateTooltip()`, where you'd return null if `this.state.category !== 'none'`